engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/hydration/NadelHydrationFieldsBuilder.kt

```diff
@@ -81,16 +77,15 @@ internal object NadelHydrationFieldsBuilder {
         val underlyingObjectType = service.underlyingSchema.getObjectType(underlyingTypeName)
             ?: error("No underlying object type")
 
-        return instructions.asSequence()
-            .flatMap { it.actorInputValueDefs }
-            .map { it.valueSource }
-            .filterIsInstance<NadelHydrationActorInputDef.ValueSource.FieldResultValue>()
-            .map { valueSource ->
+        return instructions
+            .asSequence()
+            .flatMap { it.sourceFields }
+            .map {
```

Comment:
Changed to use `sourceFields` instead of looking at arguments for source fields, as discussed in the blueprint factory comment.